### PR TITLE
adding main.py file

### PR DESCRIPTION
**Pull request by lanaHamayel5**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Task 1
+import json
+
+score = 0
+
+# Functions prototype
+def checkAnswer(enteredAnswer, correctAnswer):
```
Comment (WaseemSayara) [inline main.py]: The function name should be `check_answer`

File: main.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    global score  
+    if enteredAnswer.lower() == correctAnswer.lower():
+        score += 1  # Increase score for a correct answer
+        return "correct"
+    else:
+        score -= 1  # Decrease score for a wrong answer
```
Comment (WaseemSayara) [inline main.py]: This is a logical error, if the answer is wrong then we do nothing

File: main.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    print("Options:")
+    for option in q_value['options']:
+        print(f"{option}")
+    print()  # Adds a newline for separation
+    
+    userAnswer = input("Please, Enter the correct choice: ")
+    result = checkAnswer(userAnswer, q_value['answer'])
+    print(f"Your answer is {result}\n")
+    print()  # Adds an extra newline for separation
+    numOfquestions+=1
```
Comment (WaseemSayara) [inline main.py]: `num_of_questions` or `questions_count`

**Review by WaseemSayara — CHANGES_REQUESTED** [review]
First of all, very nice job, it works perfectly, and i really like your attention to details.
But I would like you to make some modifications,

1. Use multiple choice [1) .. , 2)..., 3)...] ask the user to enter a number.
2. use `f __name__ == “__main__”:` read more here: https://www.geeksforgeeks.org/what-does-the-if-__name__-__main__-do/

File: main.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# Task 1: Simple Quiz implementaion
+import json
+ 
+values = [4, 1, 1, 1, 3, 4, 1]
```
Comment (WaseemSayara) [inline main.py]: This is not needed
please find another way to check for the result.
Hint: compare values

File: main.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+   
+    # print(f"index = {index}")
+    if int(entered_answer) == values[index]:        
+        score += 1  # Increase score for a correct answer
+        index+=1
+        return "correct"
+       
+    else:
+        index+=1
+        return "wrong"
```
Comment (WaseemSayara) [inline main.py]: you can do something better ans easier here

File: main.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            print("Invalid input. Please enter a number.")
+           
+               
+def main():  
+    global score  
+    # Reading file  
+    with open('data.json', 'r') as file:
+        data = json.load(file)
+    # to make sure that the data was read correctlly
+    # print(data)
```
Comment (WaseemSayara) [inline main.py]: remove these two lines

File: main.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+ 
+    j=1
```
Comment (WaseemSayara) [inline main.py]: use meaningful name

File: main.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+            j+=1
+        j=1
+        print()
+       
+        choice = get_valid_choice()
+       
+        result = check_answer(choice, q_value['answer'])
+        print(f"Your answer is {result}\n")
+        print()  
+        questions_count+=1
```
Comment (WaseemSayara) [inline main.py]: Code duplication, you make something more dynamic without code duplication.

File: main.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+   
+    #Present the final answer  
+    print(f"Your final score = {score}/{questions_count}  ")
+ 
+if __name__ == "__main__":
+    main()
```
Comment (WaseemSayara) [inline main.py]: Whener you see this red icon, it means there should be an empty line at the end of the file

**Review by WaseemSayara — COMMENTED** [review]
Much cleaner, but we can still make better

File: data.json [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+          "2",
+          "8"
+        ],
+        "answer": "18"
+      }
+    }
+  }
+}
+
+
```
Comment (WaseemSayara) [inline data.json]: only one line is enough

File: main.py [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+            questions_count+=1
+        
+    #Present the final answer  
+    print(f"Your final score = {score}/{questions_count}  ")
+ 
+if __name__ == "__main__":
+    main()
+    
+    
+
```
Comment (WaseemSayara) [inline main.py]: keep 1 line

File: main.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+def main():  
+    global score  
+    # Reading file  
+    with open('data.json', 'r') as file:
+        data = json.load(file)
+    
+    num_of_options=1
+    questions_count = 0
+   
+    for question_type in ['sport', 'maths']:
```
Comment (WaseemSayara) [inline main.py]: this can be replaced with 
`for question_type in data['quiz']:`
This is more dynamic

File: main.py [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        print(f"{question_type.capitalize()} Questions:")
+        for q_key, q_value in data['quiz'][question_type].items():
```
Comment (WaseemSayara) [inline main.py]: `q_key` is not used

File: main.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+# Functions prototype
+def check_answer(entered_answer, correct_answer):
+    global score
+    if int(entered_answer) == correct_answer:        
+        score += 1  # Increase score for a correct answer
+        return "correct"
+       
+    else:
+        return "wrong"
```
Comment (WaseemSayara) [inline main.py]: lets compare strings, and make the function returm bool: (Troe or False) and increase the score in the main.
this way we get rid of the `global score`, and make it simpler.

File: main.py [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+def get_valid_choice():
+    while True:
+        try:
+            user_answer = int(input("Please, Enter the correct choice (1, 2, 3, 4): "))
+            if user_answer in [1, 2, 3, 4]:
+                return user_answer
+            else:
+                print("Invalid choice. Please enter a number between 1 and 4.")
+        except ValueError:
+            print("Invalid input. Please enter a number.")
```
Comment (WaseemSayara) [inline main.py]: Can we make it more dynamic, for example what if one of the questions has only two options?

**Review by WaseemSayara — COMMENTED** [review]
Very nice, added few comments for better readabilty and to make it more dynamic

**Review by waseem-sayara — APPROVED** [review]

**Review by WaseemSayara — APPROVED** [review]

**Pull request merged**